scripts/test_adv_IM.py

```diff
@@ -14,23 +24,81 @@
         "adv_im_model", nargs="+", type=str, help="list of adv_IM models that ran"
     )
 
+    parser.add_argument(
+        "--components",
+        default=[const.Components.c000.str_value, const.Components.c090.str_value],
+        nargs="+",
+        choices=list(const.Components.iterate_str_values()),
+        help="list of component that ran",
+    )
+
     args = parser.parse_args()
 
+    return args
+
+
+def main(sim_dir, adv_im_model, components, exit=True):
+
     failed = []
-    for model in args.adv_im_model:
-        csv_path = os.path.join(args.sim_dir, "IM_calc/{}.csv".format(model))
+
+    for model in adv_im_model:
+        # check if csv exist
+        csv_path = os.path.join(sim_dir, "IM_calc/{}.csv".format(model))
+        im_dir = get_im_calc_dir(sim_dir)
         # check file exist first to prevent crash
         try:
             df = pd.read_csv(csv_path)
         except FileNotFoundError:
             failed.append(model)
             print("csv for {} does not exist".format(model))
```

Comment:
`continue` so you don't need the following block to all be indented?

Reply:
whats the issue of it being indented? I think logic is more clear if following try, except, else?

Reply:
With all the extra (unneeded) indents, you go up to 8 indentation levels, which I think is excessive

Reply:
If you are unsure why these are unneeded - have a quick chat to James. He's got an idea of how you don't need as many indentation levels.